iOS_SDK/OneSignalSDK/Source/OSMessagingController.m

```diff
@@ -697,7 +699,27 @@ - (void)addKeySceneToWindow:(UIWindow*)window {
     }
 }
 
+- (void)dynamicTriggerCompleted:(NSString *)triggerId {
+    [OneSignal onesignal_Log:ONE_S_LL_VERBOSE message:[NSString stringWithFormat:@"messageDynamicTriggerCompleted called with triggerId: %@", triggerId]];
+    [self makeRedisplayMessagesAvailableWithTriggers:@[triggerId]];
+}
+
+- (void)makeRedisplayMessagesAvailableWithTriggers:(NSArray<NSString *> *)triggerIds {
```

Comment:
this logic is deferring a bit from Android, do we want to come with a common logic?

Reply:
After working on it I think that not duplicating the state in the redisplayedMessages might be better what do you think?

Reply:
I know that can be confusing having 2 different lists, although I see that having the redisplay messages separated can optimize the iteration. Because on worst case redisplay list will be the same length as the iam list, but in the best case redisplay iams will be empty. But is also true that there are not huge lists, or they shouldn't be, so there is not that much impact on the performance. So if you see that is more readable to only iterate iams list then I'm ok with removing the redisplay iteration for trigger change check

Reply:
Right I think having it more readable and not trying to synchronize the state is the better approach here. Sorry for making you implement it in the first place!

Reply:
ok agree then, will make the change under major release branch